dissect/target/plugins/apps/productivity/msoffice.py

```diff
@@ -0,0 +1,413 @@
+
+        manifest_path: Path = self.target.resolve(manifest_path_str.removesuffix("|vstolocal"), user_sid)
+        manifest_parser = ClickOnceDeploymentManifestParser(manifest_path, self.target, user_sid)
+        return manifest_parser.find_codebases(manifest_path)
+
+    def _parse_web_addin_manifest(self, manifest_path: Path) -> OfficeWebAddinRecord:
+        """Parses a web addin manifest."""
+
+        ns = {"": "http://schemas.microsoft.com/office/appforoffice/1.1"}
+
+        manifest_tree: Element = ElementTree.fromstring(manifest_path.read_text("utf-8-sig"))
+
+        source_location_elements = manifest_tree.findall(".//SourceLocation", ns)
+        source_locations = [source_location.get("DefaultValue") for source_location in source_location_elements]
+
+        display_name_element = manifest_tree.find(".//DisplayName", ns)
+        display_name = display_name_element.get("DefaultValue") if display_name_element is not None else None
+
+        return OfficeWebAddinRecord(
+            name=display_name,
+            manifest=manifest_path,
+            version=manifest_tree.findtext(".//Version", namespaces=ns),
+            provider_name=manifest_tree.findtext(".//ProviderName", namespaces=ns),
+            source_locations=filter(None, source_locations),
+            modification_time=manifest_path.stat().st_mtime,
+        )
+
+    def _office_install_root(self, component: Literal["Word", "Excel"]) -> str:
+        """Return the installation root for a office component"""
+
+        # Typically, all components share the same root.
+        # Curiously enough, the "Common" component has no InstallRoot defined.
+        key = f"HKLM\\{self.OFFICE_KEY}\\16.0\\{component}\\InstallRoot"
+        try:
+            return self.target.registry.value(key, "Path").value
+        except RegistryError:
+            return self.OFFICE_DEFAULT_ROOT
+
+    @lru_cache(16)
+    def _machine_startup_folders(self) -> Iterable[str]:
+        """Return machine-scoped office startup folders"""
+
+        yield fsutil.join(self._office_install_root("Word"), "STARTUP", alt_separator="\\")
+        yield fsutil.join(self._office_install_root("Excel"), "XLSTART", alt_separator="\\")
+        yield fsutil.join(self._office_install_root("Word"), "Templates", alt_separator="\\")
+        yield fsutil.join(self._office_install_root("Word"), "Document Themes", alt_separator="\\")
+
+    def _parse_plugin_status(self, addin: KeyCollection) -> NativePluginStatus | None:
+        """Parse the registry value which controls if the add-in autostarts.
+
+        See https://learn.microsoft.com/en-us/visualstudio/vsto/registry-entries-for-vsto-add-ins?view=vs-2022#LoadBehavior # noqa: E501
+        """
+
+        load_behavior = addin.value("LoadBehavior", None).value
+
+        if load_behavior is None:
+            return None
+        elif load_behavior == 0:
+            return NativePluginStatus(loaded=False, load_behavior=LoadBehavior.Manual)
```

Comment:
Wouldn't a `Dataclass` be a better approach for this? With the initial `loaded` and `load_behaviour` set to `None`. 

This this way you always return something, and avoids the `if None` checking in the `native()` function and potentially somewhere else.

Reply:
That would imply that either one of `loaded` and `load_behavior` can be ``None``, which cannot happen; either both values are present, or they are both absent.

So I think it is better to leave as-is.

Reply:
That's true, but you end up setting them to `None` anyway in the `OfficeNativeAddinRecord` anyway. So I thought the suggested approach would be a bit nicer.

Reply:
I appreciate the suggestion and agree that it is  clunky at the call-site.

However, I think it comes down to Python lacking an optional chaining operator such as in `status?.loaded`, 
and I would rather not trade invariants to compensate for that.